### ISSUE TITLE
DKIM signing does not work if letter is sent from KMail

### ISSUE THREAD

**Issue by pfactum** [vstakhov/rspamd#1750]
### Classification (Please choose *one* option):

* [ ] Crash/Hang/Data loss
* [ ] WebUI/Usability
* [ ] Serious bug
* [x] Other bug
* [ ] Feature
* [ ] Enhancement

### Reproducibility (Please choose *one* option):

* [X] Always
* [ ] Sometimes
* [ ] Rarely
* [ ] Unable
* [ ] I didn’t try
* [ ] Not applicable

### Rspamd version:

Latest rspamd-1.6 git branch.

### Operation system, CPU, memory and environment:

Up-to-date Arch Linux, x86_64, postfix.

### Description (Please provide a descriptive summary of the issue):

Google Mail as well as verifier.port25.com fail to verify DKIM signature of a letter signed by rspamd if this letter was sent from KMail. If the letter was sent from another mail app (tested with K-9 Mail, Roundcube and "mail" CLI on server), verification passes without issues.

Logs are given below.

### Compile errors (if any):

N/A

### Steps to reproduce:

Send a letter from KMail 5.5.2 via SMTP.

### Expected results:

Good letter (sent): https://gist.github.com/3fd7df1db2d233fcb9c88155f2510be2
rspamd log: https://gist.github.com/f7d114ccac76d04469a17abc56f45e64
DKIM verification result letter: https://gist.github.com/ddcf6ecf5ed7fab04af97f38b00383b5

### Actual results:

Bad letter (sent): https://gist.github.com/038722e60e45e6338b56b4e0a1f521d2
rspamd log: https://gist.github.com/0dd3f066773157886130f5d2e577b483
DKIM verification result letter: https://gist.github.com/0736f4739ccbd281c0159ec767108910

### Debugging information (see details [here](https://rspamd.com/doc/faq.html#how-to-figure-out-why-rspamd-process-crashed)):

N/A

### Configuration:

```
$ cat dkim_signing.conf
selector = "dkim-20170712";
path = "/etc/rspamd/dkim/$domain.$selector.key";
```

### Additional information:

IPs/hostnames in logs are sanitized.

Verification letter holds a copy of original letter at the very bottom. Note that good and bad original letter have different order of header fields. Maybe, this confuses rspamd?

Will provide any additional info if needed.

**Comment by pfactum** [comment]
If `Content-Transfer-Encoding` header is forced to `8bit` value in KMail, signing works okay.

**Comment by pfactum** [comment]
After deeper investigation it seems that if `Content-Transfer-Encoding` is set to `7Bit` (even if `Content-Type` is `us-ascii`), it breaks things. If it has a value of `8bit` or `base64`, (and `Content-Type: UTF-8` ) it works okay.

**Comment by pfactum** [comment]
`mail-tester.com` also complains.

Verifying signed message with `rspamc` shows no error.

**Comment by pfactum** [comment]
Removing `Content-Transfer-Encoding` header in KMail at all also makes verification successful, although UTF-8 messages are not shown for recipient properly.

**Comment by pfactum** [comment]
Fixed by 0d5eb86635b738d55f703b33336ae978effa540b, many thanks.
